Backend/cabal.project

```diff
@@ -36,3 +36,12 @@ packages:
   test
   lib/producer
 
+-- 6 parallel jobs by default
+-- you can increase this number if on a more powerful CPU
+jobs: 6
+
+-- ### DEVELOPMENTENT FLAGS ###
+-- Note: you may uncomment these while doing local development for faster compile & build times
+
+-- optimization: 0
+-- flags: +Local
```

Comment:
Why not leave this uncommented by default? AFAICT, this won't affect Nix builds, anyway. haskell-flake uses cabal.project for `packages`; everything else is ignored.

Reply:
We can if we want to. 
I just deduced from the previous PR that introduced this flag and conditional logic that we would want these to be opt-in or explicitly set by the user.

Reply:
Let's first confirm why we actually need it to be conditional. @shreyaxh @hemantmangla ?

Reply:
We had conditional logic because we want optimized code when we are in prod, but in local our concern is faster build time only.

Reply:
Ok, so we can, in fact, leave this uncommented since prod builds (built via Nix) won't use `cabal.project` (non-packages) settings anyway.

Reply:
Aside: I do wonder why we are doing all of this, instead of simply disabling optimizations **by default** and then re-enabling them only in Nix.

Reply:
LMK guys, to leave this on or off by default.
Resolved the other issues.